src/hb-ot-cmap-table.hh

```diff
@@ -1422,15 +1426,16 @@ struct cmap
     cmap *cmap_prime = c->serializer->start_embed<cmap> ();
     if (unlikely (!c->serializer->check_success (cmap_prime))) return_trace (false);
 
+    unsigned count = encodingRecord.len;
```

Comment:
With these changes this method has become very big and complex. It needs to be split up into helper methods to make it easier to follow what's happening. I'll recommend some places where I think things can be split off in following comments.